src/hotspot/share/jfr/support/jfrThreadLocal.hpp

```diff
@@ -342,6 +367,45 @@ class JfrThreadLocal {
   void set_thread_blob(const JfrBlobHandle& handle);
   const JfrBlobHandle& thread_blob() const;
 
+  // CPU time sampling
+#ifdef LINUX
+  void set_cpu_timer(timer_t timer) {
+    _has_cpu_timer = true;
+    _cpu_timer = timer;
+  }
+
+  void unset_cpu_timer() {
+    _has_cpu_timer = false;
```

Comment:
Do we really need a separate bool to understand whether the thread has a timer? Can't that information be deduced from the timer_t?

Reply:
Do we want to make hard assumptions on timer_t?

Reply:
You can have a field as a pointer: timer_t* timer and test it, timer != nullptr.

Reply:
Wouldn't I need to store it somewhere?

Reply:
You store the pointer in JfrThreadLocal, instead of the inlined timer_t struct.

One benefit is that you can fwd declare the type and need mot include its header.